test/e2e/framework/pod/utils.go

```diff
@@ -119,6 +125,7 @@ func GetLinuxLabel() *v1.SELinuxOptions {
 func GetRestrictedPodSecurityContext() *v1.PodSecurityContext {
 	return &v1.PodSecurityContext{
 		RunAsNonRoot:   pointer.BoolPtr(true),
+		RunAsUser:      pointer.Int64(1000),
```

Comment:
was this required? I thought RunAsNonRoot was sufficient

are all the callers of this function ok with running their images as uid 1000?

Reply:
It's not required as long as the image is configured to run as a non-root user, otherwise the container will fail at runtime.

It looks like `pause` (65535) and `nonroot` (1234) are the only 2 test images setting a user, so I'm inclined to keep this for now (neither would care about being run as user 1000). It would be good to update more test images to run as non-root though.

Inconsistency was just an oversight (fixed).

Reply:
I also updated the unit test to catch inconsistencies like this.